lib/template-loader.php

```diff
@@ -107,6 +107,42 @@ function gutenberg_find_template( $template_file ) {
 	if ( $template_query->have_posts() ) {
 		$template_posts            = $template_query->get_posts();
 		$_wp_current_template_post = array_shift( $template_posts );
+
+		// Build map of template slugs to their priority in the current hierarchy.
+		$slug_priorities = new stdClass();
+		foreach ( $slugs as $index => $slug ) {
+			$slug_priorities->{$slug} = $index;
+		}
+
+		// See if there is a theme block template with higher priority than the resolved template post.
+		$higher_priority_block_template_path     = null;
+		$higher_priority_block_template_priority = PHP_INT_MAX;
+		foreach ( glob( get_template_directory() . '/block-templates/*.html', 1 ) as $path ) {
```

Comment:
I think `get_stylesheet_directory()` should be considered as well to allow for child theme overrides, perhaps something like:

```suggestion
		foreach ( array_unique( array_merge( glob( get_stylesheet_directory() . '/block-templates/*.html', 1 ), glob( get_template_directory() . '/block-templates/*.html', 1 ) ) ) as $path ) {
```

Reply:
We should probably only do that if the current theme is a child theme.
```php
$block_template_files = glob( get_stylesheet_directory() . '/block-templates/*.html', 1 );
if ( is_child_theme() ) {
    $block_template_files = array_merge( $block_template_files, glob( get_template_directory() . '/block-templates/*.html', 1 ) );
}
```